subsys/bluetooth/host/gatt.c

```diff
@@ -108,6 +108,14 @@ static ssize_t read_appearance(struct bt_conn *conn,
 }
 
 #if defined (CONFIG_BT_GAP_PERIPHERAL_PREF_PARAMS)
+/* This checks if the range entered is valid */
+BUILD_ASSERT(!(CONFIG_BT_PERIPHERAL_PREF_MIN_INT > 0 &&
+	     CONFIG_BT_PERIPHERAL_PREF_MIN_INT < 6));
+BUILD_ASSERT(!(CONFIG_BT_PERIPHERAL_PREF_MAX_INT > 0 &&
+	     CONFIG_BT_PERIPHERAL_PREF_MAX_INT < 6));
+BUILD_ASSERT(CONFIG_BT_PERIPHERAL_PREF_MIN_INT >
+	     CONFIG_BT_PERIPHERAL_PREF_MAX_INT);
```

Comment:
This last BUILD_ASSERT will not work if both MIN and MAX are set to 0.

Reply:
Right, I could have fixed that with use of <= but I will change to the full range.

Reply:
Actually this isn't quite right yet: you could have MIN_INT=65535 and then MAX_INT < MIN_INT. So the check should probably be `MIN_INT == 0xffff || MIN_INT <= MAX_INT`, right?

Reply:
Right, we still need to check for 0xffff separately.